Userland/Libraries/LibWeb/Geometry/DOMMatrix.cpp

```diff
@@ -252,11 +252,14 @@ void DOMMatrix::set_f(double value)
 // https://drafts.fxtf.org/geometry/#dom-dommatrix-invertself
 JS::NonnullGCPtr<DOMMatrix> DOMMatrix::invert_self()
 {
+    bool is_invertible = m_matrix.is_invertible();
+
     // 1. Invert the current matrix.
-    m_matrix = m_matrix.inverse();
+    if (is_invertible)
+        m_matrix = m_matrix.inverse();
 
     // 2. If the current matrix is not invertible set all attributes to NaN and set is 2D to false.
```

Comment:
This spec comment seems to indicate that it's fine to divide by zero? I.e. the matrix is then set to all `NaN`s

Reply:
In theory, the fpenv control word could've been set to signal divbyzero. If we don't want to rely on that, we should do this patch. If we do want to rely on 0/0 == nan, we should remove step 2 and say "inverse() does adjunct() / determinant(), 0 determinant() implies 0 adjunct(), so inverse computes 0 / 0 for each element in that case, which does the right thing" in a comment instead, right?

Reply:
Agreed!